can/interfaces/canalystii.py

```diff
@@ -69,132 +57,355 @@ class VCI_CAN_OBJ(Structure):
+        ("AccMask", c_ulong),
+        ("Reserved", c_ulong),
+        ("Filter", c_ubyte),
+        ("Timing0", c_ubyte),
+        ("Timing1", c_ubyte),
+        ("Mode", c_ubyte),
+    ]
+
+
+class VCI_CAN_OBJ(Structure):
+    _fields_ = [
+        ("ID", c_uint),
+        ("TimeStamp", c_int),
+        ("TimeFlag", c_byte),
+        ("SendType", c_byte),
+        ("RemoteFlag", c_byte),
+        ("ExternFlag", c_byte),
+        ("DataLen", c_byte),
+        ("Data", c_byte * 8),
+        ("Reserved", c_byte * 3),
+    ]
+
+
+class VCI_BOARD_INFO(Structure):
+    _fields_ = [
+        ('hw_Version', c_ushort),
+        ('fw_Version', c_ushort),
+        ('dr_Version', c_ushort),
+        ('in_Version', c_ushort),
+        ('irq_Num', c_ushort),
+        ('can_Num', c_byte),
+        ('str_Serial_Num', c_char * 20),
+        ('str_hw_Type', c_char * 40),
+        ('Reserved', c_ushort * 4)
+    ]
+
+
+class VCI_CAN_STATUS(Structure):
+    _fields_ = [
+        ('ErrInterrupt', c_ubyte),
+        ('regMode', c_ubyte),
+        ('regStatus', c_ubyte),
+        ('regALCapture', c_ubyte),
+        ('regECCapture', c_ubyte),
+        ('regEWLimit', c_ubyte),
+        ('regRECounter', c_ubyte),
+        ('regTECounter', c_ubyte),
+        ('Reserved', c_ulong)
+    ]
+
+
+class VCI_ERR_INFO(Structure):
+    _fields_ = [
+        ('ErrCode;', c_uint),
+        ('Passive_ErrData', c_byte * 3),
+        ('ArLost_ErrData', c_byte)
+    ]
+
+
+_in_use = {}
+
+
 class CANalystIIBus(BusABC):
     def __init__(
         self,
-        channel,
-        device=0,
-        bitrate=None,
-        Timing0=None,
-        Timing1=None,
+        channel: int,
+        device: int = 0,
+        bitrate: Optional[int] = None,
+        Timing0: Optional[int] = None,
+        Timing1: Optional[int] = None,
         can_filters=None,
         **kwargs,
     ):
         """
+        CANalystII Interface
+
+        :param channel: channel number to use
+        :type channel: int
+
+        :param device: device index. If there is more then one adapter the index number will be
+            incremented by one for each additional adapter.
+        :type device: int
+
+        :param bitrate: CAN network bandwidth (bits/s).
+        :type bitrate: int, optional
+
+        :param Timing0: customize the timing register if bitrate is not specified.
+        :type Timing0: int, optional
+
+        :param Timing1: customize the timing register if bitrate is not specified.
+        :type Timing1: int, optional
 
-        :param channel: channel number
-        :param device: device number
-        :param bitrate: CAN network bandwidth (bits/s)
-        :param Timing0: customize the timing register if bitrate is not specified
-        :param Timing1:
         :param can_filters: filters for packet
         """
-        super().__init__(channel=channel, can_filters=can_filters, **kwargs)
 
-        if isinstance(channel, (list, tuple)):
-            self.channels = channel
```

Comment:
Since this is a breaking change (which I think is okay since next is a major release and it was a bug that you could only read from the first channel) it would probably be good to raise a helpful exception about multiple channels not being supported anymore.

Reply:
fair enough